Google Scholar.js

```diff
@@ -101,198 +112,96 @@ function getProfileResults(doc, checkOnly) {
 }
 
 
-function doWeb(doc, url) {
+async function doWeb(doc, url) {
+	// Determine the domain and language variant of the page.
+	let urlObj = new URL(url);
+	// Set the globals.
+	GS_CONFIG.baseURL = urlObj.origin;
+	GS_CONFIG.lang = urlObj.searchParams.get("hl") || "en";
+
 	var type = detectWeb(doc, url);
 	if (type == "multiple") {
-		if (getSearchResults(doc, true)) {
-			Zotero.selectItems(getSearchResults(doc, false), function (items) {
-				if (!items) {
-					return;
-				}
-				var ids = [];
-				for (var i in items) {
-					ids.push(i);
-				}
-				// here it is enough to know the ids and we can call scrape directly
-				scrape(doc, ids);
-			});
+		// We'll construct an appropriate "multi-scraper" async function.
+		let multiScraper;
+
+		if (getSearchResults(doc, true/* checkOnly */)) {
+			let items = await Z.selectItems(getSearchResults(doc, false));
+			if (!items) {
+				return;
+			}
+
+			multiScraper = makeGSScraper(Object.keys(items),
+				rowsFromSearchResult,
+				new URL(doc.location));
 		}
-		else if (getProfileResults(doc, true)) {
-			Zotero.selectItems(getProfileResults(doc, false), function (items) {
-				if (!items) {
-					return;
-				}
-				var articles = [];
-				for (var i in items) {
-					articles.push(i);
-				}
-				// here we need open these pages before calling scrape
-				ZU.processDocuments(articles, scrape);
-			});
+		else if (getProfileResults(doc, true/* checkOnly */)) {
+			let urls = await Z.selectItems(getProfileResults(doc, false));
+			if (!urls) {
+				return;
+			}
+			const pfName = text(doc, "#gsc_prf_in");
+
+			multiScraper = makeGSScraper(Object.keys(urls),
+				rowsFromProfile,
+				getEmulatedSearchURL(pfName), true/* expensive */);
 		}
+
+		await multiScraper(doc);
 	}
 	else {
 		// e.g. https://scholar.google.de/citations?view_op=view_citation&hl=de&user=INQwsQkAAAAJ&citation_for_view=INQwsQkAAAAJ:u5HHmVD_uO8C
-		scrape(doc, url, type);
+		await scrape(doc, url, type);
 	}
 }
 
 
-function scrape(doc, idsOrUrl, type) {
-	if (Array.isArray(idsOrUrl)) {
-		scrapeIds(doc, idsOrUrl);
-	}
-	else if (type && type == "case") {
-		scrapeCase(doc, idsOrUrl);
+// Scrape one GS entry.
+async function scrape(doc, url, type) {
+	if (type && type == "case") {
+		scrapeCase(doc, url);
 	}
 	else {
-		var related = ZU.xpathText(doc, '//a[contains(@href, "q=related:")]/@href');
-		if (!related) {
-			throw new Error("Could not locate related URL");
-		}
-		var itemID = related.match(/=related:([^:]+):/);
-		if (itemID) {
-			scrapeIds(doc, [itemID[1]]);
+		// Stand-alone "Article View" page
+		const pfName = text(doc, "#gsc_sb_ui > div > a");
+		let emuRowObj = parseArticleView(doc);
```

Comment:
Same. `Obj` is fine, but should spell out whatever `emu` is, unless this is actually about [emus](https://www.google.com/search?q=emus&tbm=isch).